### PR TITLE
Adding the Start of Navigation

### PR DESCRIPTION
**Pull request by grkalinich**
In this branch we finally introduced the **_Navigation_** feature into our program allowing us to switch between tabs on the main page and finally click into an article. The way we are using navigation within this branch is how we are planning to use it almost everywhere else (depending on what exactly we are wanting the user to see/do). 

Here is a checklist of items that occurred during this branch:
- Added a Category deserializer which allows a **Json to be converted to a C# object using Uri objects**
- Created a **tabbed view** with the start of some tabs we want to be  to be on the **HomePage** where the user starts in the app
- Deleted Files that we were able to modify and make better with the **use of CollectionViews and DataTemplates**
- **Updated from .NET Maui** 6.0 to 7.0

**And then just like any other programing update I've ever seen in my life:**
- **Fixed various bugs.**
- **Fixed performance issues.**

### CODE REVIEW

File: BITS-App/App.xaml.cs [hunk 1]
```diff
@@ -9,8 +9,10 @@ public partial class App : Application {
 	public App() {
 		InitializeComponent();
 
-		client = new HttpClient();
+        MainPage = new NavigationPage(new Pages.HomePage());
```
Comment (MandarinPine18) [inline BITS-App/App.xaml.cs]: Ah yes, we set MainPage to a NavigationPage and then proceed to throw it away _immediately_

File: BITS-App/AppShell.xaml
Comment (MandarinPine18) [inline BITS-App/AppShell.xaml:1]: I don't think we're even still using this file. We should just delete it.

File: BITS-App/AppShell.xaml.cs
Comment (MandarinPine18) [inline BITS-App/AppShell.xaml.cs:1]: Gotta delete this too

File: BITS-App/Models/PostsSet.cs
Comment (MandarinPine18) [inline BITS-App/Models/PostsSet.cs:1]: I don't think we're even using this prototype anymore either

File: BITS-App/Pages/AboutPage.xaml
Comment (MandarinPine18) [inline BITS-App/Pages/AboutPage.xaml:1]: This'll need some sprucing up, but that's for a later branch

File: BITS-App/Views/PostPage.xaml.cs
Comment (MandarinPine18) [inline BITS-App/Views/PostPage.xaml.cs:1]: This should really be in the Pages folder

**Review by MandarinPine18 — CHANGES_REQUESTED** [review]
There's quite a few minor changes, and a lot of files which aren't being used at all. We really need to delete those. I would also go back and check for formatting, because there's a heck of a lot of that which needs to be fixed. See my comments. Besides the cleanup, it's looks great! Loving the design and the snappy functionality!

**Review by MandarinPine18 — APPROVED** [review]
Alrighty it looks good now

**Comment by grkalinich** [comment]
-666... you removed the devil so that we chillin now good job :)

**Pull request merged**